### PR TITLE
Game/AI: SMART_ACTION_JUMP_TO_POS resumes path when reached jump location

### LINKED ISSUE #18760: [3.3.5/Master] Game/AI: SMART_ACTION_JUMP_TO_POS should resume path when reached jump location
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:**  smart script action SMART_ACTION_JUMP_TO_POS .

**Current behaviour:** don't resume path when reached jump location.

**Expected behaviour:** should resume path when reached jump location.

**Steps to reproduce the problem:**

1. Get the [Escape from Silverbrook ](http://www.wowhead.com/quest=12308/escape-from-silverbrook) quest for example.
2. Sit on a ducal's horse.
3. When ducal's horse reached jump location waypoint path isn't resuming.

**Branch(es):** 3.3.5 / master

**TC rev. hash/commit:** https://github.com/TrinityCore/TrinityCore/commit/b1fa8ff028fbb7469ef93f5c5612869ec9ba37ae

**TDB version:**  TDB335.62

**Operating system:**  Linux

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

### PR DESCRIPTION
**Pull request by ghost**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**
NPCs using `SMART_ACTION_JUMP_TO_POS` should resume their path when reaching their jump location.

The changes in this PR makes the action work as intended.

**Target branch(es):** 3.3.5/master

- [X] 3.3.5

**Issues addressed:** Closes #18760

**Tests performed:** (Does it build, tested in-game, etc.)

Builds and runs without issues. Tested in-game by @Eliminationzx according to dialog with @xinef1 in https://github.com/TrinityCore/TrinityCore/issues/18760#issuecomment-271087648 and below.

**Known issues and TODO list:** (add/remove lines as needed)

I don't know any, but @Eliminationzx and @xinef1 might know if there are issues.

### CODE REVIEW

**Comment by Eliminationzx** [comment]
Any feedback?

**Comment by Rushor** [comment]
works great.

**Comment by tkrokli** [comment]
Any TC dev who could confirm if the PR content is acceptable?

File: src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp [hunk 1]
```diff
@@ -142,8 +142,6 @@ void EffectMovementGenerator::Finalize(Unit* unit)
     {
         if (Unit* victim = unit->GetVictim())
             unit->GetMotionMaster()->MoveChase(victim);
-        else
-            unit->GetMotionMaster()->Initialize();
```
Comment (jackpoz) [inline src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp]: any idea about what could this break ? also why do you need to remove this to fix the smart script ?

Comment (tkrokli) [inline src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp]: ping @Eliminationzx - pls reply.

Comment (xinef1) [inline src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp]: MotionMaster::Initialize drops all other movement generators and initializes default one (set in DB for creatures). This behavior is completely wrong, there is no reason to drop all other movement generators when this one is finalized.

Comment (tkrokli) [inline src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp]: Thanks, that would explain why pathing breaks for creatures reaching jump location.

**Comment by jackpoz** [comment]
Let's see what we break

**Pull request merged**